poetry/utils/exporter.py

```diff
@@ -59,5 +60,13 @@ def _export_requirements_txt(
             line += "\n"
             content += line
 
-        with filepath.open("w", encoding="utf-8") as f:
-            f.write(decode(content))
+        self._output(content, cwd, output)
+
+    def _output(self, content, cwd, output=None):  # type: (str, Path, str) -> None
+        decoded = decode(content)
+        if output is not None:
```

Comment:
Please write `if output is None` condition first because it's default value.
```suggestion
        if output is None:
            sys.stdout.write(decoded)
```

Reply:
Good point; fixed with commit https://github.com/sdispater/poetry/pull/1035/commits/36cb4300894f9fb6114971df1b546db31fb46f4e.